elastic/apm-server
# [7.17] update beats to update the docker base image

## Motivation/summary
Beats changed the docker base image from `centos` to `ubuntu 20.04`. Updating beats is required to also apply the change to APM Server.

This pulls in breaking changes from libbeat related to idxmgmt; https://github.com/elastic/beats/commit/84bf434017a88c9d1a1860d711bfcfa4abe3c5fb#diff-701b51cb205dec9db43f1a3cd736b11c6123705c17be1786570c264bcb6c93f8 adds `IsAlias()` to the `IndexSelector` Interface. It enables the output handler to require an `alias` instead of an `index` when ingesting data. This PR introduces the missing function, but for now always returns `false` for the `IsAlias()` check. Even when ILM is enabled, APM Server still needs to write to indices for `sourcemaps` and `onboarding`. Therefore the newly introduced check cannot simply be leveraged by APM at this point. 




## Checklist




## How to test these changes

* Build docker image `DEV=true SNAPSHOT=true PLATFORMS=linux/amd64 TYPES=docker mage package`
* Start apm-server docker image and exec into it
* Ensure the base image is Ubuntu 20.04
```
apm-server@bcf8c1e2426b:~$ cat /etc/redhat-release /etc/lsb-release /etc/debian_version
cat: /etc/redhat-release: No such file or directory
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.3 LTS"
bullseye/sid
```


## Related issues
related to https://github.com/elastic/beats/issues/29255
